test/Mono.Linker.Tests/TestCasesRunner/AssemblyChecker.cs

```diff
@@ -493,6 +493,11 @@ protected virtual void VerifyCustomAttributes (ICustomAttributeProvider src, ICu
 			var expectedAttrs = GetExpectedAttributes (src).ToList ();
 			var linkedAttrs = FilterLinkedAttributes (linked).ToList ();
 
+			for (int i=0; i < linkedAttrs.Count; i++) {
```

Comment:
I can make this optional if the addition of attributes is too much, but the codegen size savings are going to motivate it being the default for Xamarin when possible.

Reply:
My comment here is with regards to how tests should work.  If I look at a test file, I should be able to understand the changes the linker is expected to make.  Quietly filtering out an attribute injected by the linker makes it so that one can no longer understand all of the changes the linker will make.

Whether or not this new option is on or off is a separate discussion.

Reply:
It's easy enough for me to just add a set of these tests with it disabled and a set with it enabled. I'll do that instead.

Reply:
I agree with Mike this is not the correct way to filter the attribute and in reality we want exact opposite